### PR TITLE
Central API object.

### PR DESCRIPTION
**Pull request by p16i**
I think we should have an object that responsible for fetching data from backend.

### CODE REVIEW

File: src/assets/js/lib/api.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import querystring from 'querystring';
+
+const api = module.exports = {};
+
+const API_URL = 'http://localhost:3100';
```
Comment (p16i) [inline src/assets/js/lib/api.js]: @rapee do you know how to get this value from config file?

Comment (rapee) [reply src/assets/js/lib/api.js]: This:
```js
import conf from './config';
const API_URL = conf.get('service.api.url');
```

File: src/assets/js/lib/api.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  let url = api._buildEndpoint(
+    'summaries',
+    {
+      start_date: start,
+      end_date: end,
+      organization: org
+    }
+  );
+
+  $.get(url,cb,'json');
```
Comment (rapee) [inline src/assets/js/lib/api.js]: Where dose this `$` come from?

Let's use Fetch API. See https://davidwalsh.name/fetch.

```js
import fetch from 'isomorphic-fetch'; // for now

fetch(url, options)
.then(response => response.json())
.then(data => {
  // parse or filter, etc.
  return data;
});
```

**Comment by rapee** [comment]
Good idea. Do you mean you want to use this on server or on client?

We could have both actually.

File: src/assets/js/lib/api.js (hunk 1 above)
Comment (p16i) [reply src/assets/js/lib/api.js]: thanks!

**Pull request closed without merging**